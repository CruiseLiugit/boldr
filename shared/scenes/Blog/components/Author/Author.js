/* @flow */
import React from 'react';
import styled from 'styled-components';
import Link from 'react-router/lib/Link';
import Avatar from 'react-md/lib/Avatars';

const AuthorInfo = styled.ul`
  list-style-type: none;
  padding-left: 0;

`;

const AvatarName = styled.li`
  display: inline-block;
  vertical-align: middle;
  &:first-child {
    margin-right: 15px;
  }
`;

const Author = (props: { username: String, avatar_url: String, last_name: String, first_name: String }) => {
  const authorName = <Link to={ `/profiles/${props.username}`}>{ props.username }</Link>;
  return (
    <div className="boldr-post__sidebar-author">
      <AuthorInfo>
        <AvatarName><Avatar src={ props.avatar_url } role="presentation" /> </AvatarName>
        <AvatarName>{ authorName }</AvatarName>
      </AuthorInfo>
      </div>
  );
};

export default Author;
