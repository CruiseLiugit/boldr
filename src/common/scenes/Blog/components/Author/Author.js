/* @flow */
import React from 'react';
import Avatar from 'material-ui/Avatar';
import List from 'material-ui/List/List';
import ListItem from 'material-ui/List/ListItem';

const Author = (props: { display_name: String, avatar_url: String }) => {
  return (
    <div className="post__sidebar-author">
        <List>
          <ListItem
            disabled
            leftAvatar={
              <Avatar src={ props.avatar_url } />
            }
          >
            { props.first_name } { props.last_name }
          </ListItem>

        </List>

      </div>
  );
};

export default Author;
