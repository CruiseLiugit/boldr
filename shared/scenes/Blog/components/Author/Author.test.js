import React from 'react';
import { createStore } from 'redux';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Author from './Author';

describe('<Author />', () => {
  const wrapper = shallow(<Author avatar_url={ 'abcd' } username="hey" />);
  it('renders <Author /> without breaking', () => {
    expect(wrapper.find('div').length).toBe(1);
    expect(wrapper.instance().props.username).toBe('hey');
  });
  it('renders snapshot', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
