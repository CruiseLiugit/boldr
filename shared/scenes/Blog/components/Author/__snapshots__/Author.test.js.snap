exports[`<Author /> renders snapshot 1`] = `
<div
  className="boldr-post__sidebar-author">
  <styled.ul>
    <styled.li>
      <Avatar
        role="presentation"
        src="abcd"
        suffixes={
          Array [
            "red",
            "pink",
            "purple",
            "deep-purple",
            "indigo",
            "blue",
            "light-blue",
            "cyan",
            "teal",
            "green",
            "light-green",
            "lime",
            "yellow",
            "amber",
            "orange",
            "deep-orange",
            "brown",
            "grey",
            "blue-grey",
          ]
        } />
       
    </styled.li>
    <styled.li>
      <Link
        onlyActiveOnIndex={false}
        style={Object {}}
        to="/profiles/hey">
        hey
      </Link>
    </styled.li>
  </styled.ul>
</div>
`;
